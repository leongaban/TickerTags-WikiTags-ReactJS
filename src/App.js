import React from 'react'

export const User = () => <div>user</div>
export default class App extends React.Component {

    constructor(props) {
        super(props)
        this.state = {
            active: false
        }
    }

	render() {
		return (
			<div className="container body-container">
                { this.props.children }
                { this.state.active && <User/> }
			</div>
		);
	}
}
