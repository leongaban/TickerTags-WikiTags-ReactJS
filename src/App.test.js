import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import App from './App'
import { User } from './App'

const users_empty = {};
const AppComponent = shallow(<App user={users_empty}/>);

describe('<App />', () => {
    it('renders', () => {
        const tree = toJson(AppComponent);
        expect(tree).toMatchSnapshot(AppComponent);
    });

    it('displays a User component if active', () =>{
        AppComponent.setState({ 'active': true });
        expect(AppComponent.find(User).length).toBe(1);
    });

    it('Does not display the User component if inactive', () => {
        AppComponent.setState({ 'active': false });
        expect(AppComponent.find(User).length).toBe(0);
    });
});