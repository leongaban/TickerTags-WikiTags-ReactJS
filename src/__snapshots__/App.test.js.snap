// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
<div
  className="container body-container"
/>
`;
